src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -584,18 +584,19 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
   -->
   <Target Name="_GetRestoreTargetFrameworksOutput"
-    DependsOnTargets="_GetRestoreProjectStyle"
+    DependsOnTargets="_GetRestoreProjectStyle;_GetRestoreTargetFrameworkOverride"
     Returns="@(_RestoreTargetFrameworksOutputFiltered)">
 
     <PropertyGroup>
       <_RestoreProjectFramework/>
+      <TargetFrameworkToBeUsed Condition=" '$(_TargetFrameworkOverride)' == '' ">$(TargetFrameworks)</TargetFrameworkToBeUsed>
```

Comment:
Should this have a private name like `$(_TargetFrameworkToBeUsed)`, if it's only used inside this one target?